### DISCUSSION TITLE
Option to not automatically add columns when model is fully defined using FluentMappingBuilder

### DISCUSSION THREAD

**Discussion by jonreis** [linq2db/linq2db#4623] [Q&A] [answered]
Hi, I just started looking at linq2db and I have a question I was hoping someone could answer for me. I looked at the code, but could not find the answer myself.

Here is my situation. I have some POCO classes that were generated using EF 6.0. The generated Entity code is placed into the main entity.cs file and additional properties/functions are added to another file (partial class of the generated Entity). I am hoping to ignore the additional properties in the partial file.

In the samples I have looked at, one needs to explicit use IsNotColumn() to mark a column as not backed by the database. If I am fully specifying the mapping from the database, is it possible to skip the code that automatically adds other columns to the Model? According to the docs sample:

"In this example we configured only three properties and one association. We let Linq To DB to infer all other properties as columns with same name as property."

But, once I have fully set the model up using the fluent interface, I don't want anything inferred. Is there a setting for this?

Appreciate any help.

1 Hour later...
I might have figured it out. It looks like I just don't call .Build() on the FluentMappingBuilder. Is that correct?

**Comment by MaceWindu** [comment, marked as answer]
1. yes, Build() call is required
2. if you define your entity using fluent mapper, it get https://github.com/linq2db/linq2db/blob/master/Source/LinqToDB/Mapping/TableAttribute.cs#L19 assigned with `IsColumnAttributeRequired = true` option. If you don't change this option - only explicitly declared columns mapped (without table attribute it works like all columns mapped if not disabled explicitly).

**Comment by jonreis** [comment]
@MaceWindu thank you very much for taking the time to respond. That is exactly what I was looking for.
